graphs/adapter-jgrapht/src/main/kotlin/com/anaplan/engineering/azuki/graphs/adapter/jgrapht/check/HasShortestPathCheck.kt

```diff
@@ -9,20 +9,28 @@ import org.jgrapht.graph.DefaultEdge
 
 class HasShortestPathCheck<V>(
     private val graphName: String,
-    private val from: V,
-    private val to: V,
-    private val shortestPath: List<V>
+    private val shortestPath: List<V>,
 ) : JGraphTCheck, GetShortestPathBehaviour() {
 
-    private val fullShortestPath by lazy {
-        listOf(from) + shortestPath + to
+    private val from = try {
+        shortestPath.first()
+    } catch (e: NoSuchElementException) {
+        error("hasShortestPath must has at least three arguments: graphName, startVertex, endVertex")
     }
+    private val to = shortestPath.last()
 
     override fun check(env: ExecutionEnvironment) =
-        checkEqual(fullShortestPath, env.get<V, List<V>>(graphName) {
+        checkEqual(shortestPath, env.get<V, List<V>>(graphName) {
+            if (shortestPath.size < 2) {
+                error("hasShortestPath must has at least three arguments: graphName, startVertex, endVertex")
+            }
             val pathAlg = DijkstraShortestPath(this as Graph<V, DefaultEdge>)
             val path = pathAlg.getPath(from, to)
-            path.vertexList
+            if (path == null) {
+                emptyList()
+            } else {
+                path.vertexList
+            }
```

Comment:
❓ I don't get the point of this. If `shortestPath` is supposed to have length at least 2, it will never be empty, so checking if it is equal to `null` or to `emptyList` will always have the same result.